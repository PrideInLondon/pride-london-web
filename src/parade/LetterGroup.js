import React from 'react'
import PropTypes from 'prop-types'

import { Heading, GroupsContainer, Wrapper } from './LetterGroup.styles'

const LetterGroup = ({ children, letter }) => {
  return (
    <Wrapper id={`parade-group-${letter}`}>
      <Heading>
        <h4>{letter.toUpperCase()}</h4>
      </Heading>
      <GroupsContainer>{children}</GroupsContainer>
    </Wrapper>
  )
}

LetterGroup.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]).isRequired,
  letter: PropTypes.string.isRequired,
}

export default LetterGroup
