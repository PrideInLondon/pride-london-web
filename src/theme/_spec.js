import { screenSizes, breakpoints } from './theme'

it.each`
  screen         | breakpointIndex
  ${'mobile'}    | ${0}
  ${'tablet'}    | ${1}
  ${'desktop'}   | ${2}
  ${'desktopHD'} | ${3}
`('successfully assigns $screen screen size', ({ screen, breakpointIndex }) => {
  expect(screenSizes[screen]).toBe(breakpoints[breakpointIndex])
})
