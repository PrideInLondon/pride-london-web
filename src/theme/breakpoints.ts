export const sm = 375
export const md = 768
export const lg = 1024
export const xl = 1440
export const nav = 1360

export interface Breakpoints {
  sm: string
  md: string
  lg: string
  xl: string
  nav: string
}

export const breakpoints: string[] & Breakpoints = Object.assign(
  [`${sm}px`, `${md}px`, `${lg}px`, `${xl}px`],
  {
    sm: `${sm}px`,
    md: `${md}px`,
    lg: `${lg}px`,
    xl: `${xl}px`,
    nav: `${nav}px`,
  }
)
