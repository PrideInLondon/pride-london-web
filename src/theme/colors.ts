export const colors = {
  black: '#000000',
  darkGrey: '#5B5B5B',
  mediumGrey: '#E1E1E1',
  lightGrey: '#EFEFEF',
  white: '#FFFFFF',
  yellow: '#FFD95E',
  tomato: '#FF5B44',
  pink: '#ED2C6E',
  fuscia: '#FF88F1',
  electricPurple: '#BC01FF',
  darkPurple: '#660C63',
  indigo: '#2D2F7F',
  darkIndigo: '#20215A',
  greyBlue: '#5770D1',
  beachBlue: '#2C77E5',
  skyBlue: '#2ACFFF',
  bondiBlue: '#00A3BD',
  darkCyan: '#008296',
  eucalyptusGreen: '#2CDA9D',
  lemonGreen: '#C3D30E',
  green: '#34C30A',
  amber: '#FFCA00',
  red: '#D0021B',
  gold: '#FFE16C',
  silver: '#E0DFDC',
  bronze: '#B59035',
  mexicanPink: '#E6007E',
  raisinBlack: '#272727',
}
