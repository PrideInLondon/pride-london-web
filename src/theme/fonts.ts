export const fonts = {
  title: 'Poppins, sans-serif',
  body: 'Roboto, sans-serif',
}
