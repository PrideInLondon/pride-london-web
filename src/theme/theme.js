const theme = {
  colors: {
    black: '#000000',
    darkGrey: '#5B5B5B',
    mediumGrey: '#E1E1E1',
    lightGrey: '#EFEFEF',
    white: '#FFFFFF',
    yellow: '#FFD95E',
    tomato: '#FF5B44',
    pink: '#ED2C6E',
    fuscia: '#FF88F1',
    electricPurple: '#BC01FF',
    darkPurple: '#660C63',
    indigo: '#2D2F7F',
    greyBlue: '#5770D1',
    beachBlue: '#2C77E5',
    skyBlue: '#2ACFFF',
    bondiBlue: '#00A3BD',
    eucalyptusGreen: '#2CDA9D',
    lemonGreen: '#C3D30E',
    green: '#34C30A',
    amber: '#FFCA00',
    red: '#D0021B',
  },
  fonts: {
    title: 'Poppins, sans-serif',
    body: 'Roboto, sans-serif',
  },
  space: [0, 5, 10, 15, 20],
  breakpoints: [375, 768, 1024, 1440],
  navBreakpoint: 1275,
}

export default theme
