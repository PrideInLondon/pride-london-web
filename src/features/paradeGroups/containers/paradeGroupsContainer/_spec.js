import React from 'react'
import { shallow, mount } from 'enzyme'
import FilterButton from '../../../../components/filterButton'
import LetterGroup from '../../components/letterGroup'
import theme from '../../../../theme/theme'
import ParadeGroupsContainer from '.'
import 'jest-styled-components'

describe(ParadeGroupsContainer.name, () => {
  const paradeGroups = [
    { name: '23 group', id: '0', category: ['Arts'] },
    { name: 'A group', id: '1', category: ['Arts'] },
    { name: 'B group', id: '2', category: ['Business'] },
  ]

  const categories = [
    {
      title: 'All groups',
      hexColour: '#FFD95E',
    },
    {
      title: 'Arts',
      hexColour: '#FF5B44',
    },
    {
      title: 'Business',
      hexColour: '#ED2C6E',
    },
  ]

  it('should render correctly', () => {
    const wrapper = shallow(
      <ParadeGroupsContainer
        paradeGroups={paradeGroups}
        categories={categories}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should filter parade groups by category Business', () => {
    const instance = mount(
      <ParadeGroupsContainer
        paradeGroups={paradeGroups}
        categories={categories}
      />
    )
    const businessCategoryFilterButton = instance.find(FilterButton).last()

    businessCategoryFilterButton.simulate('click')
    setTimeout(() => {
      expect(instance.find(LetterGroup).length).toBe(1)
      instance.unmount()
    })
  })

  it('should set active letter styles on scroll to letter section', () => {
    const instance = mount(
      <ParadeGroupsContainer
        paradeGroups={paradeGroups}
        categories={categories}
      />
    )
    const bLetter = instance.find('a[href="#parade-group-b"]')
    const bSection = instance.find('div[id="parade-group-b"]')
    bSection.simulate('scroll')
    setTimeout(() => {
      expect(bLetter).toHaveStyleRule('color', theme.colors.indigo)
      unmount(instance)
    })
  })
})
