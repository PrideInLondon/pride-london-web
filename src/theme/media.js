import { css } from 'styled-components'
import theme from './theme'

export const media = {
  // Min width media queries
  mobile: (...args) => css`
    @media (min-width: ${theme.breakpoints[0]}px) {
      ${css(...args)};
    }
  `,
  tablet: (...args) => css`
    @media (min-width: ${theme.breakpoints[1]}px) {
      ${css(...args)};
    }
  `,
  desktop: (...args) => css`
    @media (min-width: ${theme.breakpoints[2]}px) {
      ${css(...args)};
    }
  `,

  desktopHD: (...args) => css`
    @media (min-width: ${theme.breakpoints[3]}px) {
      ${css(...args)};
    }
  `,
  // Custom breakpoint for navigation
  nav: (...args) => css`
    @media (min-width: ${theme.navBreakpoint}px) {
      ${css(...args)};
    }
  `,

  // Max width media queries
  mobileMax: (...args) => css`
    @media (max-width: ${theme.breakpoints[0] - 1}px) {
      ${css(...args)};
    }
  `,
  tabletMax: (...args) => css`
    @media (max-width: ${theme.breakpoints[1] - 1}px) {
      ${css(...args)};
    }
  `,
  desktopMax: (...args) => css`
    @media (max-width: ${theme.breakpoints[2] - 1}px) {
      ${css(...args)};
    }
  `,

  desktopHDMax: (...args) => css`
    @media (max-width: ${theme.breakpoints[3] - 1}px) {
      ${css(...args)};
    }
  `,
  // Custom breakpoint for navigation
  navMax: (...args) => css`
    @media (max-width: ${theme.navBreakpoint - 1}px) {
      ${css(...args)};
    }
  `,
}
