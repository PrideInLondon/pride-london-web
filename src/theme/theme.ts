import { breakpoints } from './breakpoints'
import { mediaQueries } from './mediaQueries'
import { colors } from './colors'

const theme = {
  colors,
  fonts: {
    title: 'Poppins, sans-serif',
    body: 'Roboto, sans-serif',
  },
  space: [0, 5, 10, 15, 20],
  breakpoints,
  mediaQueries,
  navBreakpoint: 1360,
}

export default theme
