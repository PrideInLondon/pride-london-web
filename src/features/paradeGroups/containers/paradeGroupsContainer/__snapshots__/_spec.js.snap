// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParadeGroupsContainer should render correctly 1`] = `
<Fragment>
  <ParadeGroupsFilterContainer
    categories={
      Array [
        Object {
          "hexColour": "#FFD95E",
          "title": "All groups",
        },
        Object {
          "hexColour": "#FF5B44",
          "title": "Arts",
        },
        Object {
          "hexColour": "#ED2C6E",
          "title": "Business",
        },
      ]
    }
    handleFilterClick={[Function]}
    selectedFilter={
      Object {
        "hexColour": "#FFD95E",
        "title": "All groups",
      }
    }
  />
  <styles__LetterContainer>
    <styles__ScrolledLetters>
      <LetterLink
        isActive={false}
        isDisabled={true}
        key="#"
        letter="#"
      />
      <LetterLink
        isActive={false}
        isDisabled={false}
        key="a"
        letter="a"
      />
      <LetterLink
        isActive={false}
        isDisabled={false}
        key="b"
        letter="b"
      />
      <LetterLink
        isActive={false}
        isDisabled={true}
        key="c"
        letter="c"
      />
      <LetterLink
        isActive={false}
        isDisabled={true}
        key="d"
        letter="d"
      />
      <LetterLink
        isActive={false}
        isDisabled={true}
        key="e"
        letter="e"
      />
      <LetterLink
        isActive={false}
        isDisabled={true}
        key="f"
        letter="f"
      />
      <LetterLink
        isActive={false}
        isDisabled={true}
        key="g"
        letter="g"
      />
      <LetterLink
        isActive={false}
        isDisabled={true}
        key="h"
        letter="h"
      />
      <LetterLink
        isActive={false}
        isDisabled={true}
        key="i"
        letter="i"
      />
      <LetterLink
        isActive={false}
        isDisabled={true}
        key="j"
        letter="j"
      />
      <LetterLink
        isActive={false}
        isDisabled={true}
        key="k"
        letter="k"
      />
      <LetterLink
        isActive={false}
        isDisabled={true}
        key="l"
        letter="l"
      />
      <LetterLink
        isActive={false}
        isDisabled={true}
        key="m"
        letter="m"
      />
      <LetterLink
        isActive={false}
        isDisabled={true}
        key="n"
        letter="n"
      />
      <LetterLink
        isActive={false}
        isDisabled={true}
        key="o"
        letter="o"
      />
      <LetterLink
        isActive={false}
        isDisabled={true}
        key="p"
        letter="p"
      />
      <LetterLink
        isActive={false}
        isDisabled={true}
        key="q"
        letter="q"
      />
      <LetterLink
        isActive={false}
        isDisabled={true}
        key="r"
        letter="r"
      />
      <LetterLink
        isActive={false}
        isDisabled={true}
        key="s"
        letter="s"
      />
      <LetterLink
        isActive={false}
        isDisabled={true}
        key="t"
        letter="t"
      />
      <LetterLink
        isActive={false}
        isDisabled={true}
        key="u"
        letter="u"
      />
      <LetterLink
        isActive={false}
        isDisabled={true}
        key="v"
        letter="v"
      />
      <LetterLink
        isActive={false}
        isDisabled={true}
        key="w"
        letter="w"
      />
      <LetterLink
        isActive={false}
        isDisabled={true}
        key="x"
        letter="x"
      />
      <LetterLink
        isActive={false}
        isDisabled={true}
        key="y"
        letter="y"
      />
      <LetterLink
        isActive={false}
        isDisabled={true}
        key="z"
        letter="z"
      />
    </styles__ScrolledLetters>
  </styles__LetterContainer>
  <div>
    <LetterGroup
      letter="a"
    >
      <ParadeGroup
        facebookUrl={null}
        instagramUrl={null}
        key="1"
        name="A group"
        twitterUrl={null}
        websiteUrl={null}
      />
    </LetterGroup>
    <LetterGroup
      letter="b"
    >
      <ParadeGroup
        facebookUrl={null}
        instagramUrl={null}
        key="2"
        name="B group"
        twitterUrl={null}
        websiteUrl={null}
      />
    </LetterGroup>
  </div>
</Fragment>
`;
